clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -12910,6 +12910,18 @@ def err_sycl_special_type_num_init_method : Error<
   "types with 'sycl_special_class' attribute must have one and only one '__init' "
   "method defined">;
 
+// SYCL external attribute diagnostics
+def err_sycl_attribute_invalid_linkage : Error<
+  "'clang::sycl_external' can only be applied to functions with external"
+  " linkage">;
+def err_sycl_attribute_invalid_main : Error<
+  "'clang::sycl_external' cannot be applied to the 'main' function">;
+def err_sycl_attribute_invalid_deleted_function : Error<
+  "'clang::sycl_external' cannot be applied to an explicitly deleted function">;
```

Comment:
For consistency with changes made in https://github.com/llvm/llvm-project/pull/151405 for the `sycl_kernel_entry_point` attribute, let's use `%0` to name the attribute. I also suggest we rename these diagnostics to be specific to `sycl_external`; we'll likely have other SYCL attributes in the future.
```suggestion
def err_sycl_external_invalid_linkage : Error<
  "%0 can only be applied to functions with external linkage">;
def err_sycl_external_invalid_main : Error<
  "%0 cannot be applied to the 'main' function">;
def err_sycl_external_invalid_deleted_function : Error<
  "%0 cannot be applied to an explicitly deleted function">;
```